### PR TITLE
UI: Converted data insight dropdown to checkbox dropdown

### PR DESCRIPTION
**Pull request by ShaileshParmar11**
### Describe your changes :

issue: https://github.com/open-metadata/OpenMetadata/issues/8996#issuecomment-1327618489
Added checkbox dropdown for data insight filters

#
### Type of change :

- [x] Improvement

#
### Frontend Preview (Screenshots) :
image

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

**Comment by chirag-madlani** [comment]
Can we increase padding around the button? I feel it's too less cc: @devyani-kaushik

File: openmetadata-ui/src/main/resources/ui/src/pages/DataInsightPage/DataInsightPage.component.tsx [hunk 1]
```diff
@@ -116,45 +134,91 @@ const DataInsightPage = () => {
...
+      setTierOptions((prev) => ({
+        ...prev,
+        options: prev.options.filter((value) =>
+          value.toLocaleLowerCase().includes(query.toLocaleLowerCase())
+        ),
+      }));
+    } else {
+      setTeamOptions((prev) => ({
+        ...prev,
+        options: defaultTierOptions,
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/DataInsightPage/DataInsightPage.component.tsx:178]: Why we are setting `defaultTierOptions` to TeamOptions? Is this intentional?

File: openmetadata-ui/src/main/resources/ui/src/utils/DataInsightUtils.tsx [hunk 2]
```diff
@@ -421,11 +421,10 @@ export const getGraphDataByTierType = (rawData: TotalEntitiesByTier[]) => {
...
 };
 
-export const getTeamFilter = (suggestionValues: ListValues = []) => {
-  return (suggestionValues as ListItem[]).map((suggestion: ListItem) => ({
-    label: suggestion.title,
-    value: suggestion.value,
-  }));
+export const getTeamFilter = (suggestionValues: ListValues = []): string[] => {
+  return (suggestionValues as ListItem[]).map(
+    (suggestion: ListItem) => suggestion.value
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/utils/DataInsightUtils.tsx:426]:
```suggestion
    (suggestion) => suggestion.value
```

File: openmetadata-ui/src/main/resources/ui/src/utils/DataInsightUtils.tsx [hunk 3]
```diff
@@ -421,11 +421,10 @@ export const getGraphDataByTierType = (rawData: TotalEntitiesByTier[]) => {
   };
 };
 
-export const getTeamFilter = (suggestionValues: ListValues = []) => {
-  return (suggestionValues as ListItem[]).map((suggestion: ListItem) => ({
-    label: suggestion.title,
-    value: suggestion.value,
-  }));
+export const getTeamFilter = (suggestionValues: ListValues = []): string[] => {
+  return (suggestionValues as ListItem[]).map(
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/utils/DataInsightUtils.tsx:425]: Why we are expecting `suggestionValues` to be `ListValues` and then we are casting it to `ListItem[]`?

**Review by chirag-madlani — CHANGES_REQUESTED** [review]

File: openmetadata-ui/src/main/resources/ui/src/pages/DataInsightPage/DataInsightPage.component.tsx (hunk 1 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/DataInsightPage/DataInsightPage.component.tsx:178]: Thanks for catching, it was not intentional, I have updated code

File: openmetadata-ui/src/main/resources/ui/src/utils/DataInsightUtils.tsx (hunk 2 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/utils/DataInsightUtils.tsx:426]: updated

File: openmetadata-ui/src/main/resources/ui/src/utils/DataInsightUtils.tsx (hunk 3 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/utils/DataInsightUtils.tsx:425]: suggestionValues will be ListValues and ListValues can be ListItem[], (string | number)[],
but you are correct it should be handled at page level and not in utility method! i have update the code!

File: openmetadata-ui/src/main/resources/ui/src/pages/DataInsightPage/DataInsightPage.component.tsx [hunk 4]
```diff
@@ -77,14 +80,27 @@ const DataInsightPage = () => {
   const { isAdminUser } = useAuth();
   const history = useHistory();
 
-  const [teamsOptions, setTeamOptions] = useState<SelectProps['options']>([]);
+  const [teamsOptions, setTeamOptions] = useState<TeamStateType>({
+    defaultOptions: [],
+    selectedOptions: [],
+    options: [],
+  });
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/pages/DataInsightPage/DataInsightPage.component.tsx:87]: Create an initial filter object and use it here.

File: openmetadata-ui/src/main/resources/ui/src/pages/DataInsightPage/DataInsightPage.component.tsx [hunk 5]
```diff
@@ -77,14 +80,27 @@ const DataInsightPage = () => {
...
+  const [tierOptions, setTierOptions] = useState<TierStateType>({
+    selectedOptions: [],
+    options: [],
+  });
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/pages/DataInsightPage/DataInsightPage.component.tsx:91]: Same here.

File: openmetadata-ui/src/main/resources/ui/src/pages/DataInsightPage/DataInsightPage.component.tsx (hunk 4 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/DataInsightPage/DataInsightPage.component.tsx:87]: I think it is fine for such simple data

File: openmetadata-ui/src/main/resources/ui/src/pages/DataInsightPage/DataInsightPage.component.tsx (hunk 5 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/DataInsightPage/DataInsightPage.component.tsx:91]: same here.

**Comment by chirag-madlani** [comment]
> Can we increase padding around the button? I feel it's too less cc: @devyani-kaushik

@ShaileshParmar11

**Review by chirag-madlani — APPROVED** [review]

**Comment by ShaileshParmar11** [comment]
> > Can we increase padding around the button? I feel it's too less cc: @devyani-kaushik
> 
> @ShaileshParmar11

Ill take care of it, in the next PR of design improvement thanks

**Pull request merged**